Mathlib/Analysis/ODE/PicardLindelof.lean

```diff
@@ -10,397 +10,586 @@ import Mathlib.Topology.MetricSpace.Contracting
+
+We actually prove a more general version of this theorem for the existence of local flows. If there
+is some `r ≥ 0` such that `L * max (tmax - t₀) (t₀ - tmin) ≤ a - r`, then for every
+`x ∈ closedBall x₀ r`, there exists a (local) solution `α x` with the initial condition `α t₀ = x`.
+In other words, there exists a local flow `α : E → ℝ → E` defined on `closedBall x₀ r` and
+`Icc tmin tmax`.
+
+The proof relies on demonstrating the existence of a solution `α` to the following integral
+equation:
+$$\alpha(t) = x_0 + \int_{t_0}^t f(\tau, \alpha(\tau))\,\mathrm{d}\tau.$$
+This is done via the contraction mapping theorem, applied to the space of Lipschitz continuous
+functions from a closed interval to a Banach space. The needed contraction map is constructed by
+repeated applications of the right hand side of this equation.
+
+## Main definitions and results
+
+* `picard f t₀ x₀ α t`: the Picard iteration, applied to the curve `α`.
+* `IsPicardLindelof`: the structure holding the assumptions of the Picard-Lindelöf theorem.
+* `IsPicardLindelof.exists_eq_hasDerivWithinAt`: the existence theorem for local solutions to
+  time-dependent ODEs.
+* `IsPicardLindelof.exists_forall_mem_closedBall_eq_hasDerivWithinAt_Icc`: the existence theorem for
+  local flows to time-dependent vector fields.
 
 ## Implementation notes
 
-In order to split the proof into small lemmas, we introduce a structure `PicardLindelof` that holds
-all assumptions of the main theorem. This structure and lemmas in the `PicardLindelof` namespace
-should be treated as private implementation details. This is not to be confused with the `Prop`-
-valued structure `IsPicardLindelof`, which holds the long hypotheses of the Picard-Lindelöf
-theorem for actual use as part of the public API.
-
-We only prove existence of a solution in this file. For uniqueness see `ODE_solution_unique` and
-related theorems in `Mathlib/Analysis/ODE/Gronwall.lean`.
+* The structure `FunSpace` and theorems within this namespace are implementation details of the
+  proof of the Picard-Lindelöf theorem and are not intended to be used outside of this file.
+* Some sources, such as Lang, define `FunSpace` as the space of continuous functions from a closed
+  interval to a closed ball. We instead define `FunSpace` here as the space of Lipschitz continuous
+  functions from a closed interval. This slightly stronger condition allows us to postpone the usage
+  of the completeness condition on the space `E` until the application of the contraction mapping
+  theorem.
+* We have chosen to formalise many of the real constants as `ℝ≥0`, so that the non-negativity of
+  certain quantities constructed from them can be shown more easily. When subtraction is involved,
+  especially note whether it is the usual subtraction between two reals or the truncated subtraction
+  between two non-negative reals.
+* We only prove the existence of a solution in this file. For uniqueness, see `ODE_solution_unique`
```

Comment:
Perhaps reword the first sentence as "in this file, we only prove the existence of a solution"?